tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -575,6 +576,104 @@ public:
    Result_t &PartialUpdate(unsigned int slot) { return *fGraphs[slot]; }
 };
 
+class FillTGraphAsymmErrorsHelper : public ROOT::Detail::RDF::RActionImpl<FillTGraphAsymmErrorsHelper> {
+public:
+   using Result_t = ::TGraphAsymmErrors;
+
+private:
+   std::vector<::TGraphAsymmErrors *> fGraphAsymmErrors;
+
+public:
+   FillTGraphAsymmErrorsHelper(FillTGraphAsymmErrorsHelper &&) = default;
+   FillTGraphAsymmErrorsHelper(const FillTGraphAsymmErrorsHelper &) = delete;
+
+   FillTGraphAsymmErrorsHelper(const std::shared_ptr<::TGraphAsymmErrors> &g, const unsigned int nSlots)
+      : fGraphAsymmErrors(nSlots, nullptr)
+   {
+      fGraphAsymmErrors[0] = g.get();
+      // Initialize all other slots
+      for (unsigned int i = 1; i < nSlots; ++i) {
+         fGraphAsymmErrors[i] = new TGraphAsymmErrors(*fGraphAsymmErrors[0]);
```

Comment:
I just realized that these `new` won't be cleaned up if we throw during the event loop (for example from the `Exec` below). It is a more generic problem than just for TGraphAsymmErrors though, we'll have to address it separately

Reply:
I agree that should be imporved - in particular, there is a similar allocation for `HIST` and `TGraph` as well, hence deserves a new issue.

Should I mark this as resolved here?

Reply:
can you please create an issue for this and assign it to me? then mark it as resolved